atlassian_jwt_auth/contrib/server/helpers.py

```diff
@@ -32,25 +31,31 @@ def _requires_asap(verifier, auth, parse_jwt_func, build_response_func,
         if verify_issuers_func is not None:
             verify_issuers_func(asap_claims, issuers)
         asap_claim_holder.asap_claims = asap_claims
-    except requests.exceptions.HTTPError as e:
+    except PublicKeyRetrieverException as e:
+        if e.status_code != 404:
+            # Any error other than "not found" is a problem and should be dealt
+            # with elsewhere
+            raise
         # Couldn't find key in key server
         message = 'Unauthorized: Invalid key'
         exception = e
-    except (requests.exceptions.ConnectionError,
-            PublicKeyRetrieverException) as e:
-        message = 'Unauthorized: Backend server connection error'
-        exception = e
     except jwt.exceptions.InvalidIssuerError as e:
-        message = 'Unauthorized: Invalid token issuer'
+        message = 'Forbidden: Invalid token issuer'
         exception = e
     except jwt.exceptions.InvalidTokenError as e:
         # Something went wrong with decoding the JWT
         message = 'Unauthorized: Invalid token'
         exception = e
     if message is not None:
         logger = logging.getLogger(__name__)
-        logger.error(message,
+        logger.debug(message,
                      extra={'original_message': str(exception)})
-        return build_response_func(message, status=401, headers={
-                              'WWW-Authenticate': 'Bearer'})
+        if message.startswith('Unauthorized:'):
```

Comment:
Maybe default the response headers to be `Unauthorized` and then check for the `Forbidden` case (or the other way around if you want).

Reply:
That would allow an unexpected message to sneak in without throwing an error.

Currently, a bad message will get logged (if debugging is turned on) and then the server will 500 since `kwargs` isn't set. Using a default `kwargs` would allow typos and other unrecognized messages through to the client with no errors.

Reply:
Fair enough.